modules/dkan/dkan_datastore/modules/dkan_datastore_api/dkan_datastore_api.module

```diff
@@ -1123,3 +1123,180 @@ function dkan_datastore_api_debug($data_select) {
   $sql = str_replace('\n', ' ', $sql);
   return (string) $sql;
 }
+
+/**
+ * Implements hook_services_resources_alter().
+ */
+function dkan_datastore_api_services_resources_alter(&$resources, $endpoint) {
+  if (isset($resources['node']['operations']['create'])) {
+    $resources['node']['operations']['create']['callback'] =
+      "_dkan_datastore_api_resource_create";
+  }
+
+  if (isset($resources['node']['operations']['update'])) {
+    $resources['node']['operations']['update']['callback'] =
+      "_dkan_datastore_api_resource_update";
+  }
+}
+
+/**
+ * Services operation callback: node creation.
+ */
+function _dkan_datastore_api_resource_create($request) {
```

Comment:
Does not check access control

Reply:
I believe the services module handles authentication and access, but we can check.